### PR TITLE
Updated dependency to govdpa v0.1.4

### PR DESCRIPTION
**Pull request by lmilleri**
A new release of [govdpa](https://github.com/k8snetworkplumbingwg/govdpa) library has been published (v.0.1.4), introducing new functionalities like add/remove vDPA devices and also significant changes to the public interface.

The current changes to the sriov-network-device-plugin are required to import govdpa v.0.1.4

### CODE REVIEW

**Comment by lmilleri** [comment]
Hi, can anyone please help with the following:
- workflow running approval
- "LGTM analysys: Javascript" is failing. Is it a real problem?
- I noticed there are a lot of unwanted changes in the vendor directory, due to automatic indentation. Is it ok to keep it?

/cc @zshi-redhat

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (adrianchiris) [inline .vscode/settings.json:1]: remove this file please

File: pkg/netdevice/netInfoProviders.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 )
 
 /*
-   rdmaInfoProvider provides the RDMA information
+rdmaInfoProvider provides the RDMA information
```
Comment (adrianchiris) [inline pkg/netdevice/netInfoProviders.go:28]: indentation changes are not related to this PR. can you either:

1. move them to separate PR
2. move them to a separate commit in this PR

File: .vscode/settings.json (hunk 1 above)
Comment (lmilleri) [reply .vscode/settings.json]: done

File: pkg/netdevice/netInfoProviders.go (hunk 2 above)
Comment (lmilleri) [reply pkg/netdevice/netInfoProviders.go]: Removed indentation changes from this PR

**Comment by lmilleri** [comment]
This commit https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pull/450/commits/a0aa5ef28c890da117d22e95c0618978d72664a7 wasn't planned to be included, checking my rebase ...

**Comment by lmilleri** [comment]
@adrianchiris Please review or add relevant people in cc

File: pkg/utils/vdpa_provider.go [hunk 3]
```diff
@@ -25,5 +27,14 @@ func GetVdpaProvider() VdpaProvider {
...
+	// the govdpa library requires the pci address to include the "pci/" prefix
+	fullPciAddr := "pci/" + pciAddr
+	vdpaDevices, err := vdpa.GetVdpaDevicesByPciAddress(fullPciAddr)
+	if err != nil {
+		return nil, err
+	}
+	if len(vdpaDevices) == 0 {
+		return nil, fmt.Errorf("no vdpa device associated to pciAddress %v", pciAddr)
+	}
+	return vdpaDevices[0], nil
```
Comment (adrianchiris) [inline pkg/utils/vdpa_provider.go:39]: so there may be multiple VDPA devices per PCI addr ? and we currently just care of the first one ?

**Comment by adrianchiris** [comment]
just one minor question on my side.
otherwise LGTM.

@lmilleri can you rebase this PR ?

File: pkg/utils/vdpa_provider.go (hunk 3 above)
Comment (bn222) [reply pkg/utils/vdpa_provider.go:39]: Maybe add a logging message that we are ignoring all but the first

**Review by adrianchiris — APPROVED** [review]

File: pkg/netdevice/vdpa.go [hunk 4]
```diff
@@ -50,6 +51,19 @@ func (v *vdpaDevice) GetType() types.VdpaType {
...
 }
 
+func (v *vdpaDevice) GetParent() string {
+	return v.VdpaDevice.Name()
+}
+
+func (v *vdpaDevice) GetPath() string {
+	path, err := v.ParentDevicePath()
+	if err != nil {
+		glog.Infof("%s - No path for vDPA device found: %s", v.Name(), err)
```
Comment (SchSeba) [inline pkg/netdevice/vdpa.go:61]: nit: for the error you should use %v

File: pkg/utils/vdpa_provider.go [hunk 5]
```diff
@@ -25,5 +28,18 @@ func GetVdpaProvider() VdpaProvider {
...
-	return vdpa.GetVdpaDeviceByPci(pciAddr)
+	// the govdpa library requires the pci address to include the "pci/" prefix
+	fullPciAddr := "pci/" + pciAddr
+	vdpaDevices, err := vdpa.GetVdpaDevicesByPciAddress(fullPciAddr)
+	if err != nil {
+		return nil, err
+	}
+	numVdpaDevices := len(vdpaDevices)
+	if numVdpaDevices == 0 {
+		return nil, fmt.Errorf("no vdpa device associated to pciAddress %v", pciAddr)
```
Comment (SchSeba) [inline pkg/utils/vdpa_provider.go:39]: nit: change %v to %s as this is a string

File: pkg/utils/vdpa_provider.go [hunk 6]
```diff
@@ -25,5 +28,18 @@ func GetVdpaProvider() VdpaProvider {
...
+	}
+	if numVdpaDevices > 1 {
```
Comment (SchSeba) [inline pkg/utils/vdpa_provider.go:41]: Is this a valid use-case?
should we always return the first one?

**Review by SchSeba — COMMENTED** [review]
thanks for the contribution the PR is almost ready to small nits and one question

File: pkg/utils/vdpa_provider.go (hunk 6 above)
Comment (lmilleri) [reply pkg/utils/vdpa_provider.go:41]: Adrian asked the same question, please check comments above. Basically, there is only one vdpa device per pci-address in SRIOV. Anyway, the govdpa interface is more general and future proof for when sub-functions will be supported.

File: pkg/utils/vdpa_provider.go (hunk 5 above)
Comment (lmilleri) [reply pkg/utils/vdpa_provider.go]: done

File: pkg/netdevice/vdpa.go (hunk 4 above)
Comment (lmilleri) [reply pkg/netdevice/vdpa.go]: done

File: pkg/utils/vdpa_provider.go (hunk 6 above)
Comment (SchSeba) [reply pkg/utils/vdpa_provider.go:41]: Good that we ask the same questions :P

**Review by SchSeba — APPROVED** [review]
/lgtm

**Review by adrianchiris — APPROVED** [review]
